packages/material-ui/src/ButtonBase/Ripple.js

```diff
@@ -26,23 +37,17 @@ function Ripple(props) {
       [classes.ripplePulsate]: pulsate,
     },
     className,
+    rippleClasses.root,
   );
 
-  const rippleStyles = {
```

Comment:
Could you revert this change?

Reply:
Are you suggesting we keep the inline styles in addition to the new class?

Reply:
Do you confirm that we only need https://github.com/mui-org/material-ui/pull/15503#issuecomment-487217244 to solve the problem?

Reply:
I would rather use inline style than the style function.

Reply:
Oh, I see what you are saying.  No, that resolves the need to use the index option on makeStyles, but it would not resolve my issue without the rest of the changes.  I have to run my code on websites with crazy garbage like `span { width: 200px !Important; }` set in their stylesheets.  So I have to clean that up in my namespace with a reset including things like `.myNamespace * { width: auto !important; }`.

What is the aversion to moving to a style function, performance?  Just seems like it would be nice to solve the CSP issue and allow any sort of preprocessing to run on these styles that users would like.

Reply:
Thanks for the extra detail, I better understand the problem now. There is close to no performance implication in this case, but it will once it's made a systematic solution. We might be able to use CSS variables in v5 (#14420). I have reverted the change of this file.
Did you consider using the shadow DOM?

Reply:
I would love to be using shadow DOM, it is perfect for my use case.  I may implement it for most users again soon (past attempted this in MUI 0.x and ran into too many issues).

That said, unfortunately we have a ton of IE11 and Edge users.  The polyfill basically does what I am doing with my jss plugins, namespaces all of the CSS.  So in those browsers I will still have this issue.

I understand not wanting to pull in this change, I will probably just patch it locally for now.  Would you mind giving some insight into where you think I will hit a performance issue when changing the other cases of inline style usage?

Reply:
I can imagine that supporting IE 11 is important for you, shadow DOM is not an option. Dropping IE 11 would also enable CSS variables and probably solve the issue here at the same time.

I'm curious why is dynamically creating an iframe is not an option for you? We do it for a few of our demos in the documentation, e.g. the Drawers' demos. If you are familiar with Stripe, it's how they increase the security of their web input widget.

Using style functions is slower. However, it would unlock a lot of interesting features for Material-UI. I'm happy to push this further but in a different pull request. Removing the dead CSS rules is already an improvement :).

Reply:
This discussion echos with the one of #15499 as it requires the usage of style functions in the core components.